### PR TITLE
Update tags definition, add tests for xinspect

### PR DESCRIPTION
**Pull request by tharun571**

### CODE REVIEW

File: CMakeLists.txt
Comment (vgvassilev) [inline CMakeLists.txt:201]: The name should be more generic. xeus_cpp_config is probably what we need.

Comment (tharun571) [reply CMakeLists.txt:201]: That's what I tried previously and the file was overridden. Had a conversation about this with @mcbarton and the way to go about saving the definitions was creating a new file.

File: CMakeLists.txt
Comment (mcbarton) [inline CMakeLists.txt:35]: @tharun571 Probably just set the variables here, and move the install part to the install section in the cmakelists.txt file.

**Review by mcbarton — APPROVED** [review]

File: CMakeLists.txt
Comment (vgvassilev) [reply CMakeLists.txt:197]: We should merge both files into one and it will work. Note that we are writing to `xeus_cpp_config` and that should probably change, too.

Comment (mcbarton) [reply CMakeLists.txt]: @vgvassilev I have helped @tharun571 to combine the config files and it is ready for another review

File: include/xeus-cpp/xeus_cpp_config.hpp.in
Comment (vgvassilev) [inline include/xeus-cpp/xeus_cpp_config.hpp.in:15]: We should configure the versions too. The current version should be defined in the calling cmake. Elsewhere we have it in a VERSION file which is read by cmake. @JohanMabille, what is your take here?

File: CMakeLists.txt
Comment (mcbarton) [inline CMakeLists.txt:37]: @tharun571 You probably want to set the version number the way we do in CppInterOp with a version file (see https://github.com/compiler-research/CppInterOp/pull/296/files for more details). It will allow us to label something as a version in development.

Comment (mcbarton) [reply CMakeLists.txt:37]: The version file for xeus-cpp would currently have 0.5.0;dev as we are currently developing version 0.5.0 and no longer on version 0.4.0

Comment (mcbarton) [reply CMakeLists.txt:37]: You won't need to add a XeusCppConfigVersion.cmake.in to be clear

**Review by vgvassilev — COMMENTED** [review]
Looks good to me but I cannot take that decision without @JohanMabille.

**Comment by mcbarton** [comment]
> Looks good to me but I cannot take that decision without @JohanMabille.

@vgvassilev There is a bug at the moment. When the config header file is configured it still has ;dev in the patch variable . Me and @tharun571 have been discussing. I can't see why. Do you see why?

**Comment by vgvassilev** [comment]
> > Looks good to me but I cannot take that decision without @JohanMabille.
> 
> @vgvassilev There is a bug at the moment. When the config header file is configured it still has ;dev in the patch variable . Me and @tharun571 have been discussing. I can't see why. Do you see why?

Do we parse the list properly the way we do in CppInterOp?

**Comment by mcbarton** [comment]
> > > Looks good to me but I cannot take that decision without @JohanMabille.
> > 
> > 
> > @vgvassilev There is a bug at the moment. When the config header file is configured it still has ;dev in the patch variable . Me and @tharun571 have been discussing. I can't see why. Do you see why?
> 
> Do we parse the list properly the way we do in CppInterOp?

In an identical way as I linked the PR where we parsed it in CppInterOp as a guide. For some reason that isn't obvious to me the patch number still has ;dev .

**Comment by mcbarton** [comment]
This ;dev doesn't happen in CppInterOp, as it was the whole reason we parsed it in this way to avoid the ;dev in patch version number, and allowing finding a version number correctly.

**Comment by vgvassilev** [comment]
Well, something must be different.

**Comment by mcbarton** [comment]
> Well, something must be different.

@vgvassilev A solution to the patch version issue has been found and the PR is ready for review by @JohanMabille

File: include/xeus-cpp/xeus_cpp_config.hpp.in
Comment (JohanMabille) [inline include/xeus-cpp/xeus_cpp_config.hpp.in:15]: This should not be configured by an external tool. This makes the cpp code depend on the building tools. This should be reverted to the previous behavior which was the correct one.

File: CMakeLists.txt
Comment (JohanMabille) [inline CMakeLists.txt:41]: See the comment in `xeus_cpp_config.hpp.on`.

File: CMakeLists.txt
Comment (JohanMabille) [inline CMakeLists.txt:180]: This should be removed (see the comment in `xeus_cpp_config.hpp.in`).

File: CMakeLists.txt
Comment (JohanMabille) [inline CMakeLists.txt:455]: Why removing this? It is required for regular unix-like system-wide installs.

**Review by JohanMabille — CHANGES_REQUESTED** [review]
~~What was the motivation for changing the logic of version and tag definitions?~~

EDIT: I just saw Vassil's comment, sorry for my late reply;

File: include/xeus-cpp/xeus_cpp_config.hpp.in
Comment (JohanMabille) [reply include/xeus-cpp/xeus_cpp_config.hpp.in]: Sorry I missed this comment. I think the opposite, the code should not depend on the building tool.

File: CMakeLists.txt
Comment (JohanMabille) [reply CMakeLists.txt]: We can add a "XEUS_CPP_LABEL" or anything else to handle the "dev" part of the version (when there is one). But as said previously, I am strongly opposed to having cmake injecting the version in the cpp code.

File: CMakeLists.txt
Comment (tharun571) [reply CMakeLists.txt:455]: Rearranged it according to the code structure. Didn't remove per say.

File: CMakeLists.txt
Comment (vgvassilev) [inline CMakeLists.txt:27]: Why do we need to put these in the cache?

Comment (tharun571) [reply CMakeLists.txt:27]: It was there in the previous code. CACHE just turns up in the cmake config tool as setting you can set. The initial commit for it was made by @mvassilev

Comment (vgvassilev) [reply CMakeLists.txt:27]: Ok. We can set them as normal variables.

**Review by vgvassilev — APPROVED** [review]
LGTM, let's wait for @JohanMabille final say.

File: CMakeLists.txt
Comment (JohanMabille) [inline CMakeLists.txt:34]: I think using `target_compile_definition` would be a better alternative than configuring hpp files from cmake.
However, let's merge it for now to have a reference to test against.

**Review by JohanMabille — APPROVED** [review]

**Pull request merged**
